File: backend/app/schemas/document.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.db.models import DocumentStatus, DocumentType


class DocumentResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    document_type: DocumentType
    title: str
    content_markdown: str
    version: int
    status: DocumentStatus
    created_by_agent_id: Optional[uuid.UUID]
    approved_by: Optional[str]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DocumentSummary(BaseModel):
    """Lightweight list item — omits content_markdown."""
    id: uuid.UUID
    project_id: uuid.UUID
    document_type: DocumentType
    title: str
    version: int
    status: DocumentStatus
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/document.ts
export type DocumentType =
  | "requirement_summary"
  | "gap_analysis_report"
  | "brd"
  | "fsd"
  | "user_story"
  | "architecture_design"
  | "database_design"
  | "api_spec"
  | "screen_spec"
  | "code_task_list"
  | "technical_design"
  | "code_review"
  | "devops_config"
  | "build_report"
  | "monitoring_report"
  | "test_report"
  | "test_cases"
  | "uat_script"
  | "change_impact_report"
  | "compiled_documents"
  | "project_summary"
  | "delivery_report";

export type DocumentStatus = "draft" | "review" | "approved" | "rejected" | "superseded";

export interface DocumentSummary {
  id: string;
  project_id: string;
  document_type: DocumentType;
  title: string;
  version: number;
  status: DocumentStatus;
  created_at: string;
  updated_at: string;
}

export interface Document extends DocumentSummary {
  content_markdown: string;
  created_by_agent_id: string | null;
  approved_by: string | null;
}
